### ISSUE TITLE
How to use the minibatch mode to train the SVGP

### ISSUE THREAD

**Issue by nonameusers** [GPflow/GPflow#795]
Hi, dear!
I'm trying to use the minibatch mode to train the SVGP, since the SVGP includes the the follwing Minibatch methods:    
        if minibatch_size is None:
               X = DataHolder(X)
               Y = DataHolder(Y)
        else:
            X = Minibatch(X, batch_size=minibatch_size, seed=0)
            Y = Minibatch(Y, batch_size=minibatch_size, seed=0)

However, I don't know how to use this code. Should I also use the following code to train the model?
opt = gpflow.train.ScipyOptimizer()
opt.minimize(m, maxiter=10)

Could I use the GradientDescentOptimizer of Tensorflow to train the model, instead of the ScipyOptimizer()?

Thanks very much !

**Comment by hughsalimbeni** [comment]
The `ScipyOptimizer` is only for deterministic objectives, so is not compatible with minibatches. If you try to use `ScipyOptimizer` with minibatches it wont throw an error but it will likely terminate after after a few iterations and give you a bad result. All the tensorflow optimizers are ok to use with minibatches, though. I suggest using a Adam or Adagrad first, however, as `GradientDescentOptimizer` can be hard to use and you'll find yourself with divergence if you're not very careful.
